RIOT-OS/RIOT
# cpu/stm32: add rtc_mem

### Contribution description

WIP of rtc_mem implementation for stm32, just to see if the API works fine for stm32 BOARDs as well.

### Testing procedure

Both tests/periph_rtc and tests/periph_rtt have been adapted to make use of the feature and to ensure it does not interfere with normal RTC/RTT alarms.

Need to see if those REGS do not exist on some families....

### Issues/PRs references

Depends on https://github.com/RIOT-OS/RIOT/pull/16758
